### PR TITLE
Add option for running the demo from docker container 

### PR DESCRIPTION
**Pull request by jappeace**
This allows people to run the demo from a docker
container, so that they don't have to compile the
interpreter or the ghc-wpc program.
It also makes all native dependencies available.

The gephi tool doesn't quite work at the moment
because it relies on hardware accelaration,
which I'm not sure hot to install for both nvidia
and amd users.
(but it's available in the container regardless)

I tested this ext-stg-interpreter successfully in
this environment for the game and the tsumupto demo's

### CODE REVIEW

**Comment by csabahruska** [comment]
What is your system GHC version? is it GHC 8.8.3?

**Comment by jappeace** [comment]
yes: https://github.com/grin-compiler/ext-stg-interpreter-presentation-demos/pull/1/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R45

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      xz-utils
+
+RUN cabal update && cabal install happy-1.19.12 && cabal install alex-3.2.6
+
+# 2. ghcup is required for installing a specific version of ghc systemwide.
+ARG GHCUP_BIN_DIR="/root/.ghcup/bin"
+ARG GHCUP_VERSION="0.1.17.2"
+ARG GHCUP_CHECKSUM="e9adb022b9bcfe501caca39e76ae7241af0f30fbb466a2202837a7a578607daf"
+
+ENV PATH=${PATH}:${GHCUP_BIN_DIR}
```
Comment (csabahruska) [inline Dockerfile]: it should be
`ENV PATH=${GHCUP_BIN_DIR}:${PATH}` to override system GHC.

**Comment by csabahruska** [comment]
I've added zstd zip support + mc zstd zip browsing support.

new file: [mc-zstd-support.zip](https://github.com/grin-compiler/ext-stg-interpreter-presentation-demos/files/7698669/mc-zstd-support.zip)

patch:
```diff
diff --git a/.gitmodules b/.gitmodules
index 75d8ccc..d92779c 100644
--- a/.gitmodules
+++ b/.gitmodules
@@ -1,3 +1,6 @@
 [submodule "ghc-whole-program-compiler-project"]
 	path = ghc-whole-program-compiler-project
 	url = git@github.com:grin-compiler/ghc-whole-program-compiler-project.git
+[submodule "unzip-zstd"]
+	path = unzip-zstd
+	url = git@github.com:csabahruska/unzip-zstd.git
diff --git a/Dockerfile b/Dockerfile
index 31b650a..a6ddb2d 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -5,6 +5,8 @@ RUN apt update && apt install -y pkg-config libbz2-dev
 
 RUN echo "Installing system dependencies required to compile GHC." &&\
     apt install -y \
+      libzstd-dev \
+      mc \
       autoconf \
       build-essential \
       curl \
@@ -27,7 +29,7 @@ ARG GHCUP_BIN_DIR="/root/.ghcup/bin"
 ARG GHCUP_VERSION="0.1.17.2"
 ARG GHCUP_CHECKSUM="e9adb022b9bcfe501caca39e76ae7241af0f30fbb466a2202837a7a578607daf"
 
-ENV PATH=${PATH}:${GHCUP_BIN_DIR}
+ENV PATH=${GHCUP_BIN_DIR}:${PATH}
 
 COPY ./fetch_ghcup.sh /tmp/fetch_ghcup.sh
 RUN chmod +x /tmp/fetch_ghcup.sh
@@ -59,3 +61,13 @@ WORKDIR /root/ghc-whole-program-compiler-project/ghc-wpc
 RUN ./boot
 RUN ./configure
 RUN hadrian/build-stack -j
+
+# 6. zstd .zip support
+
+# zstd support for unzip
+WORKDIR /root/unzip-zstd
+RUN make -f unix/Makefile generic_zstd && cp unzip $HOME/.local/bin
+
+# midnight commander zstd-zip browsing
+WORKDIR /root/
+RUN unzip mc-zstd-support.zip
diff --git a/mc-zstd-support.zip b/mc-zstd-support.zip
new file mode 100644
index 0000000..6e5616e
Binary files /dev/null and b/mc-zstd-support.zip differ
diff --git a/unzip-zstd b/unzip-zstd
new file mode 160000
index 0000000..fd97f08
--- /dev/null
+++ b/unzip-zstd
@@ -0,0 +1 @@
+Subproject commit fd97f0832fa48f4e8f468e9fa90f2f13b9339aa6
```

**Comment by csabahruska** [comment]
Should I add support for the gephi graph visualization tool also?
https://gephi.org/users/install/

**Comment by jappeace** [comment]
yes, you can just push into this branch I believe

**Review by csabahruska — APPROVED** [review]
Looks good! Thanks.

**Pull request merged**